appinventor/components/src/com/google/appinventor/components/runtime/Web.java

```diff
@@ -901,6 +934,21 @@ public void run() {
         } else {
           final String responseContent = getResponseContent(connection);
 
+          // Update the locally stored columns list with the contents of the
```

Comment:
@ewpatton What if a user does Get on the Web component, and then changes the Source of multiple data components later on? In that case, the current implementation would retrieve the Web component's cached resources. Otherwise, the user would have to call Get again, thus using up another request (which could be detrimental for limited request APIs). I guess for most use cases it makes sense, but adding the if condition could introduce some hassle to the user, forcing them to do things one way rather than the other.

Reply:
@ewpatton For the exception catching, I guess it also makes sense to just post the UI call before the two newly introduced lines of code, since the order here shouldn't really matter. I will move the UI call up and we will not really need to deal with more specific exceptions there (since I believe all the necessary exceptions are caught inside the inner methods, and any other exceptions should cause a crash since that should represent very undefined behavior)